src/biogeophys/FrictionVelocityMod.F90

```diff
@@ -516,26 +557,61 @@ subroutine SetRoughnessLengthsAndForcHeightsNonLake(this, bounds, &
 
          frac_veg_nosno   =>    canopystate_inst%frac_veg_nosno_patch , & ! Input:  [integer  (:)   ] fraction of vegetation not covered by snow (0 OR 1) [-]
          frac_sno         =>    waterdiagnosticbulk_inst%frac_sno_col , & ! Input:  [real(r8) (:)   ] fraction of ground covered by snow (0 to 1)
+         snomelt_accum    =>    waterdiagnosticbulk_inst%snomelt_accum_col , & ! Input:  [real(r8) (:)   ] accumulated col snow melt for z0m calculation (m H2O)       
          urbpoi           =>    lun%urbpoi                            , & ! Input:  [logical  (:)   ] true => landunit is an urban point
          z_0_town         =>    lun%z_0_town                          , & ! Input:  [real(r8) (:)   ] momentum roughness length of urban landunit (m)
          z_d_town         =>    lun%z_d_town                          , & ! Input:  [real(r8) (:)   ] displacement height of urban landunit (m)
          forc_hgt_t       =>    atm2lnd_inst%forc_hgt_t_grc           , & ! Input:  [real(r8) (:)   ] observational height of temperature [m]
          forc_hgt_u       =>    atm2lnd_inst%forc_hgt_u_grc           , & ! Input:  [real(r8) (:)   ] observational height of wind [m]
-         forc_hgt_q       =>    atm2lnd_inst%forc_hgt_q_grc             & ! Input:  [real(r8) (:)   ] observational height of specific humidity [m]
+         forc_hgt_q       =>    atm2lnd_inst%forc_hgt_q_grc           , & ! Input:  [real(r8) (:)   ] observational height of specific humidity [m]
+         z0mg_2D          =>    this%z0mg_2D_col                        & ! Input:  [real(r8) (:)   ] 2-D field of input col roughness length over ground, momentum  [m] 
+         
          )
 
     do fc = 1, num_nolakec
        c = filter_nolakec(fc)
 
        ! Ground roughness lengths over non-lake columns (includes bare ground, ground
        ! underneath canopy, wetlands, etc.)
-       if (frac_sno(c) > 0._r8) then
-          z0mg(c) = this%zsno
-       else
-          z0mg(c) = this%zlnd
-       end if
+
+       select case (z0param_method)
+       case ('ZengWang2007')
+          if (frac_sno(c) > 0._r8) then
+             if(use_z0m_snowmelt) then
```

Comment:
For code maintenance unless this option is needed for science we'd like to simplify it and not add this option for ZengWang.

@olyson or @swensosc do you have opinions on how useful this would be for running with ZengWang?

Reply:
If we simplify though we should change the buildnml so that you can NOT use this option with ZengWang.

Reply:
I'm not aware of any need for this. Might be better to simplify.

Reply:
Removed